src/app/persistence/AttributePersistence.h

```diff
@@ -43,7 +43,7 @@ class AttributePersistence
     ///
     /// Error reason for load failure is logged (or nothing logged in case "Value not found" is the
     /// reason for the load failure).
-    template <typename T, typename std::enable_if_t<std::is_arithmetic_v<T>> * = nullptr>
+    template <typename T, typename std::enable_if_t<std::is_arithmetic_v<T> || std::is_enum_v<T>> * = nullptr>
```

Comment:
we can extract this into a separate PR.

Could we have unit tests on it as well?

Reply:
Moved to a different PR #40568 and added unit test.